pd/src/components/staking.rs

```diff
@@ -211,6 +215,18 @@ impl Staking {
         // The pending delegation changes should be empty at the beginning of the next epoch.
         self.delegation_changes = Default::default();
 
+        // Set the pending reward notes on the JMT for the current block height
+        // so they can be processed by the ShieldedPool.
+        let height = self.overlay.get_block_height().await?;
+        self.overlay
+            .put_domain(
+                format!("reward_notes/{}", height).into(),
+                RewardNotes {
+                    notes: reward_notes,
+                },
+            )
+            .await;
```

Comment:
This should be wrapped up in methods on `staking::View`, like
```rust
async fn set_reward_notes(&self, height: block::Height, notes: RewardNotes) { /* ... */ }
async fn reward_notes(&self, height: block::Height) -> Result<Option<RewardNotes>> { /* ... */ }
```
so that we can separate the tree accesses (working with the paths, etc) from the app logic that uses the data, and so that we only ever define the paths in one place (the `View` trait for the component).

Also, I think the key should probably be `staking/reward_notes/{}` so that it has the same prefix as the other keys.

Reply:
Yeah I wasn't sure what prefix to use since it's shared between two components. Is the pattern going to be "prefix it with the setter of the field"?

Reply:
I think if we have a rule that we only ever define paths in a `View` trait, I think we get the answer "for free", in that we can just use the prefix of whatever component is defining that view